scipy/io/matlab/streams.pyx

```diff
@@ -318,8 +245,6 @@ def _read_string(GenericStream st, size_t n):
 cpdef GenericStream make_stream(object fobj):
     """ Make stream of correct type for file-like `fobj`
     """
-    if npy_PyFile_Check(fobj):
```

Comment:
This code block always returns a GenericStream.
I'm assume that `npy_PyFile_Check(GenericStreamInstance) == False`
otherwise the new code will return `GenericStreamInstance` while the old code would return a `GenericStream(GenericStreamInstance)`

Reply:
`npy_PyFile_Check` is an alias for `PyFile_Check` which `Return true if its argument is a PyFileObject or a subtype of PyFileObject.`

`GenericStream is not a subclass of PyFileObject` so this logic is correct.

Reply:
Thanks! That reasoning sounds right.